server/shadow/shadow_server.c

```diff
@@ -499,6 +504,13 @@ int shadow_server_parse_command_line(rdpShadowServer* server, int argc, char** a
 		if (!freerdp_settings_set_bool(settings, FreeRDP_NlaSecurity, FALSE))
 			return COMMAND_LINE_ERROR;
 	}
+
+	if (freerdp_settings_get_bool(settings, FreeRDP_NlaSecurity))
+	{
+		const char* sam = freerdp_settings_get_string(settings, FreeRDP_NtlmSamFile);
+		if (!sam)
```

Comment:
Just that this check probably prevents usage of the default `/etc/winpr/SAM` file unless its path isn't explicitly specified over `/sam-file:` argument. Is that intentional?

Reply:
hmm, true.
so need to modify this to also check for a global user database.